llvm/test/CodeGen/X86/isel-shift.ll

```diff
@@ -0,0 +1,756 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py
+; RUN: llc < %s -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,SDAG
+; RUN: llc < %s -fast-isel -fast-isel-abort=1 -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,FASTISEL
+; RUN: llc < %s -global-isel -global-isel-abort=1 -mtriple=x86_64-apple-darwin10 | FileCheck %s --check-prefixes=COMMON,GISEL
```

Comment:
Would it be safe to add i686 coverage? I'm not sure if all gisel legalization is in place or not.

Reply:
Hmm, yeah, there is a problem with `G_ICMP` and `i64` input. Probably it'd look weird: reorganizing shuffle tests with `G_ICMP` legalization.
BTW, do you mean `i686` only for `GlobalISel` or for all of them?

Reply:
Adding i686 for all of them - but if GlobalIsel is currently failing, then just the other two (with a TODO to add GlobalISel i686 tests in the future).

Reply:
Tunnel shifts are required to support `i64` shifts for `i686`. I already have  the implementation on Phabricator. So there is a todo for now.